scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/RestResponseController.java

```diff
@@ -68,19 +68,23 @@
       value = "Get metrics by run name",
       response = Json[].class,
       responseContainer = "HttpEntity")
-  @ApiResponses({@ApiResponse(code = 200, message = "Success")})
+  @ApiResponses({
+    @ApiResponse(code = 200, message = "Success"),
+    @ApiResponse(code = 404, message = "Run not found")
+  })
   public HttpEntity<byte[]> getMetricsByName(
       @PathVariable("name") @ApiParam(value = "Run name") String id) {
-    return ResponseEntity.ok()
-        .contentType(MediaType.APPLICATION_JSON_UTF8)
-        .body(
-            scheduler
-                .finished()
-                .filter(dto -> dto.getRunAlias().equals(id))
-                .findAny()
-                .map(NotificationDto::getMetrics)
-                .orElse("[]")
-                .getBytes(StandardCharsets.UTF_8));
+    byte[] response =
+        scheduler
+            .finished()
+            .filter(dto -> dto.getRunAlias().equals(id))
+            .findAny()
+            .map(NotificationDto::getMetrics)
+            .orElse("[]")
+            .getBytes(StandardCharsets.UTF_8);
+    return (response.length > 0
```

Comment:
I think `"[]".getBytes(StandardCharsets.UTF_8).length will be > 0` too, no? This is checking the length of the byte array rather than the JSON array.

You can probably return `...findAny().map(...).getBytes(...).orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND))`, similar to the `/run/{name}` endpoint